### PR TITLE
CLDR-14400 Complete Vuefication (modernization) of About Box

### PR DESCRIPTION
**Pull request by srl295**
CLDR-14400

This is meant to also be a cookbook showing what we can do.

- use a Vue-based About Box
- Basic for now, but a simple template
- Delete AboutST and code in SurveyAjax, use an /api route
- Also, add "npm run watch" in the /js directory, this will rebuild bundle.js on any change

UI: 
image

it's not necessarily a UI improvement, but shows a modern structure.

### CODE REVIEW

File: tools/cldr-apps/js/package.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
   "scripts": {
     "test": "exit 0",
     "lint": "exit 0",
-    "build": "webpack"
+    "build": "webpack",
+    "watch": "webpack watch"
```
Comment (srl295) [inline tools/cldr-apps/js/package.json:10]: this turned out to be really handy!  I have `mvn … liberty:dev` in one window, and `npm run watch` in another. Instant reload of any file, whether .js  (fraction of seconds) or .java (10s of seconds!)

File: tools/cldr-apps/js/src/views/About.vue [hunk 2]
```diff
@@ -1,5 +1,73 @@
 <template>
```
Comment (srl295) [inline tools/cldr-apps/js/src/views/About.vue:1]: comments:
- the CSS is *localized*.  Saves hunting around.
- this code simply enumerates the properties in `aboutData`.

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/About.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                    + "  \"java_vm_version\": \"15.0.1+9\",\n"
+                    + "  \"os_arch\": \"x86_64\",\n"
+                    + "  \"os_name\": \"Mac OS X\",\n"
+                    + "  \"os_version\": \"10.16\",\n"
+                    + "  \"serverInfo\": \"IBM WebSphere Liberty/21.0.0.1\",\n"
+                    + "  \"servletVersion\": \"4.0\"\n"
+                    + "}\n"
+                    + "")))})
+    public Response getAbout() {
+        Map<String,String> r = new TreeMap<>();
```
Comment (srl295) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/About.java:65]: for now it's a port (copy and paste really) of `AboutST.java` but could be more complex if needed.

File: tools/cldr-apps/js-unittest/TestCldrAbout.js [hunk 4]
```diff
@@ -30,12 +30,5 @@
         "does contain angle brackets"
       );
     });
-
-    const label = "ICU";
-    const value = json.ICU_VERSION;
-    it("should contain " + label + " and " + value, function () {
-      assert(html.indexOf(label) !== -1, label + " does occur in " + html);
-      assert(html.indexOf(value) !== -1, value + " does occur in " + html);
-    });
```
Comment (btangmu) [inline tools/cldr-apps/js-unittest/TestCldrAbout.js:39]: Would this part of the test not pass anymore?

**Review by btangmu — APPROVED** [review]

File: tools/cldr-apps/js/package.json (hunk 1 above)
Comment (btangmu) [reply tools/cldr-apps/js/package.json:10]: When running that way, can you set breakpoints and step through the code?

Comment (btangmu) [reply tools/cldr-apps/js/package.json:10]: And is that with Eclipse, or without?

Comment (srl295) [reply tools/cldr-apps/js/package.json:10]: No impact on front end experience. It means that the npm build happens automatically. This is just for the generation of bundle.js. Yes you can debug in eclipse or in browser.

File: tools/cldr-apps/js-unittest/TestCldrAbout.js (hunk 4 above)
Comment (srl295) [reply tools/cldr-apps/js-unittest/TestCldrAbout.js:39]: Correct because Gethtml doesn't fetch the old endpoint.

**Comment by btangmu** [comment]
@srl295 I just noticed CLDR_DATA_HASH etc. aren't clickable -- it would be good to restore them as links like https://github.com/unicode-org/cldr/commit/910849be69ff37687a229c6140aa5361d76c381f

**Comment by srl295** [comment]
> @srl295 I just noticed CLDR_DATA_HASH etc. aren't clickable -- it would be good to restore them as links like [910849b](https://github.com/unicode-org/cldr/commit/910849be69ff37687a229c6140aa5361d76c381f)

Certainly, I can do that as Vue side code.  Probably could even pull some of that code forwad.  I wanted to keep this PR more straightforward.

**Pull request merged**